public/app/core/navigation/parseKeyValue.ts

```diff
@@ -123,6 +126,7 @@ export function tryDecodeURIComponent(value: string): string | void {
     return decodeURIComponent(value);
   } catch (e) {
     // Ignore any invalid uri component.
+    return;
```

Comment:
Maybe we should return an empty string here and then we could get rid of `isDefined` function all together? Also, I think we should remove the `isUndefined` function on line 163 which doesn't seem to be used. 

Something to consider outside of the scope of this PR: why are most of these functions exported? @dprokop any thoughts on this?